conformance_tests/tools/debug/src/test_debug.cpp

```diff
@@ -1342,6 +1342,7 @@ void zetDebugReadWriteRegistersTest::run_read_write_registers_test(
     delete[] kernel_buffer;
 
     LOG_INFO << "[Debugger] resuming interrupted threads";
+    lzt::clear_exceptions(device, debugSession, device_threads);
```

Comment:
+1 this makes sense, would also eliminate possibility of forgetting to do this.

Reply:
okay, so i am not sure, if its good idea to clear exceptions the moment we are getting stopped threads. Would clearing exceptions impact any read/write of stopped thread's register and memory in any way?

Reply:
another thought I had, why isn't this just done in a driver as part of the zetDebugResume API itself?

Reply:
no shouldnt do it in driver. As its very much test case specific for now. We are doing it as in GDB also its done. And as per bspec, sip should clear those exceptions before resuming to application . And that discussion is already happening.

Reply:
right, my point is that it is not the responsibility of the application, in this case the test, to clear the exceptions.

Reply:
@jitendrasharma1989 did you determine that there is an issue with clearing the exceptions immediately? The only think I could think would be if that would cause the threads to resume before calling zetDebugResume.

Reply:
The driver should not be clearing exceptions.  Only debugger (GDB) should be clearing exceptions. So test needs to clear them.

There should be no issue with clearing exceptions immediately.  It wont cause thread to resume or interfere with anything.